### PR TITLE
Update Rule “ask-for-help/rule”

### PR DESCRIPTION
**Pull request by SamBWagner**
Automatically generated by Netlify CMS

### CODE REVIEW

File: rules/ask-for-help/rule.md [hunk 1]
```diff
@@ -4,14 +4,33 @@ title: Do you know when to ask for help?
...
 created: 2022-08-30T04:06:01.553Z
 guid: 113b1e11-6df4-4901-9e9f-bcc06b6fb352
 ---
 When you're working, you will encounter issues that block you from progressing, force you to do a lot of research or otherwise spend time building towards being able to complete your work. In these moments, the question arises, "When should I ask for help?"
 
+Before asking for help, make sure you've done your due-diligence and exhausted your normal avenues for unblocking yourself. Here are some examples for developers:
+
+- Try Googling it
+- Check Stack Overflow
+- Read the documentation e.g. [Microsoft Docs](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.microsoft.com%2Fen-au%2F&data=05%7C01%7CSamWagner%40ssw.com.au%7C2b80dfbf1b4c4397f32808da8aecff12%7Cac2f7c34b93548e9abdc11e5d4fcb2b0%7C0%7C0%7C637975047735355922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=EMhkvkYBn%2BCbTaN7jP%2F9kem9MM%2FkaOPSKLHe3Gej6AM%3D&reserved=0)
```
Comment (bradystroud) [inline rules/ask-for-help/rule.md:22]: Outlook safelinks are ugly 🤮

```suggestion
- Read the documentation e.g. [Microsoft Docs](https://docs.microsoft.com/en-au/)
```

Comment (pierssinclairssw) [inline rules/ask-for-help/rule.md:22]: change to https://docs.microsoft.com/en-au/

File: rules/ask-for-help/rule.md [hunk 2]
```diff
@@ -4,14 +4,33 @@ title: Do you know when to ask for help?
...
+- Explore the code e.g.
+- look at if any other pages do similar things
+- put breakpoints and check the values to see if you can figure out little bits
+- Systematically check things, check a tiny part, confirm your understanding, then move to the next part
+- Try to resolve it yourself a few times (at least 30 mins)
+
```
Comment (pierssinclairssw) [inline rules/ask-for-help/rule.md:28]: make sure these are properly indented

File: rules/ask-for-help/rule.md [hunk 3]
```diff
@@ -4,14 +4,33 @@ title: Do you know when to ask for help?
...
 2. If you spend that amount of time trying to work out the problem, you **should** ask for help
+
+Once you've determined that you need help it is vital to [draft up context about what you have done](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.ssw.com.au%2Frules%2Fdo-you-add-context-reasoning-to-your-emails&data=05%7C01%7CSamWagner%40ssw.com.au%7C2b80dfbf1b4c4397f32808da8aecff12%7Cac2f7c34b93548e9abdc11e5d4fcb2b0%7C0%7C0%7C637975047735355922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=rNlv9Yw%2B3JuEQLF%2BMmRgdOALxNoWw8%2FQv4n0hU5Ry2U%3D&reserved=0) and [provide options so they have some starting point](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.ssw.com.au%2Frules%2Fdo-you-manage-up&data=05%7C01%7CSamWagner%40ssw.com.au%7C2b80dfbf1b4c4397f32808da8aecff12%7Cac2f7c34b93548e9abdc11e5d4fcb2b0%7C0%7C0%7C637975047735355922%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=coB2z2m0aKvqUK7GmYXVjh28aFAoHgksg82QrpUHJB4%3D&reserved=0).
```
Comment (bradystroud) [inline rules/ask-for-help/rule.md:36]: Outlook safelinks are ugly 🤮
```suggestion
Once you've determined that you need help it is vital to [draft up context about what you have done](https://www.ssw.com.au/rules/do-you-add-context-reasoning-to-your-emails) and [provide options so they have some starting point](https://www.ssw.com.au/rules/do-you-manage-up).
```

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: rules/ask-for-help/rule.md (hunk 3 above)
Comment (pierssinclairssw) [inline rules/ask-for-help/rule.md:36]: change to https://www.ssw.com.au/rules/do-you-add-context-reasoning-to-your-emails
and
https://www.ssw.com.au/rules/do-you-manage-up

File: rules/ask-for-help/rule.md [hunk 4]
```diff
@@ -4,14 +4,33 @@ title: Do you know when to ask for help?
...
+  - title: Piers Sinclair
+    url: https://ssw.com.au/people/piers-sinclair
 related:
   - do-you-use-timeboxing-to-avoid-wasted-time
+  - do-you-manage-up
+  - do-you-add-context-reasoning-to-your-emails
 created: 2022-08-30T04:06:01.553Z
 guid: 113b1e11-6df4-4901-9e9f-bcc06b6fb352
 ---
 When you're working, you will encounter issues that block you from progressing, force you to do a lot of research or otherwise spend time building towards being able to complete your work. In these moments, the question arises, "When should I ask for help?"
```
Comment (pierssinclairssw) [inline rules/ask-for-help/rule.md:17]: missing line 

"This concept is critical when trying to talk with Senior developers because their time is much more valuable. If you [call them without having done any research, it will waste a lot of their time and yours](https://www.ssw.com.au/rules/investigate-your-question-for-two-minutes-before-asking-someone-on-im) and won't be a good [ROI](https://www.ssw.com.au/rules/return-on-investment)"

**Review by pierssinclairssw — CHANGES_REQUESTED** [review]

File: rules/ask-for-help/rule.md (hunk 1 above)
Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: Changed

File: rules/ask-for-help/rule.md (hunk 3 above)
Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: Changed

File: rules/ask-for-help/rule.md (hunk 1 above)
Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: Fixed

File: rules/ask-for-help/rule.md (hunk 3 above)
Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: Fixed

File: rules/ask-for-help/rule.md (hunk 4 above)
Comment (SamBWagner) [reply rules/ask-for-help/rule.md:17]: Added :)

File: rules/ask-for-help/rule.md (hunk 2 above)
Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: Done :)

File: rules/ask-for-help/rule.md [hunk 5]
```diff
@@ -4,14 +4,35 @@ title: Do you know when to ask for help?
...
+  - put breakpoints and check the values to see if you can figure out little bits
+  - Systematically check things, check a tiny part, confirm your understanding, then move to the next part
+- Try to resolve it yourself a few times (at least 30 mins)
+
+
+
 As everyone's struggles are different and everyone's way of handling pressure is different, it's hard to provide a useful metric for precisely when to ask for help. However, here's a general rule to follow:
 
 1. Find the smallest 'significant' amount of work that your project can allocate. In Scrum, this would be the smallest effort value (Extra-Small if using shirt sizes)
 2. If you spend that amount of time trying to work out the problem, you **should** ask for help
```
Comment (bradystroud) [inline rules/ask-for-help/rule.md:36]: @SamBWagner @pierssinclairssw 

I preferred the suggestion of asking for help once you exceed the estimate. 

```suggestion
As everyone's struggles are different and everyone's way of handling pressure is different, it's hard to provide a useful metric for precisely when to ask for help. However, as a general rule, if you are blocked for more than the estimated time on the PBI, ask for help.
```

File: rules/ask-for-help/rule.md [hunk 6]
```diff
@@ -4,14 +4,35 @@ title: Do you know when to ask for help?
...
+Before asking for help, make sure you've done your due-diligence and exhausted your normal avenues for unblocking yourself. This concept is critical when trying to talk with Senior developers because their time is much more valuable. If you [call them without having done any research, it will waste a lot of their time and yours](https://www.ssw.com.au/rules/investigate-your-question-for-two-minutes-before-asking-someone-on-im) and won't be a good [ROI](https://www.ssw.com.au/rules/return-on-investment). Here are some examples for developers:
+
+- Try Googling it
+- Check [Stack Overflow](https://stackoverflow.com/)
+- Read the documentation e.g. [Microsoft Docs](https://docs.microsoft.com/en-au/)
+- Explore the code e.g.
+  - look at if any other pages do similar things
+  - put breakpoints and check the values to see if you can figure out little bits
+  - Systematically check things, check a tiny part, confirm your understanding, then move to the next part
+- Try to resolve it yourself a few times (at least 30 mins)
```
Comment (bradystroud) [inline rules/ask-for-help/rule.md:29]: Capitalize all first letters
```suggestion
- Try Googling it
- Check [Stack Overflow](https://stackoverflow.com/)
- Read the documentation e.g. [Microsoft Docs](https://docs.microsoft.com/en-au/)
- Explore the code e.g.
  - Look at if any other pages do similar things
  - Put breakpoints and check the values to see if you can figure out little bits
  - Systematically check things, check a tiny part, confirm your understanding, then move to the next part
- Try to resolve it yourself a few times (at least 30 mins)
```

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: rules/ask-for-help/rule.md [hunk 7]
```diff
@@ -4,14 +4,35 @@ title: Do you know when to ask for help?
...
+
+
+
 As everyone's struggles are different and everyone's way of handling pressure is different, it's hard to provide a useful metric for precisely when to ask for help. However, here's a general rule to follow:
 
 1. Find the smallest 'significant' amount of work that your project can allocate. In Scrum, this would be the smallest effort value (Extra-Small if using shirt sizes)
```
Comment (pierssinclairssw) [inline rules/ask-for-help/rule.md:35]: I think we should reword this to be a bit easier to understand

e.g.

"If the time worked on your PBI has exceeded the effort and you aren't getting anywhere then you should seek help"

**Review by pierssinclairssw — DISMISSED** [review]

File: rules/ask-for-help/rule.md (hunk 5 above)
Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: I personally don't prefer it as I believe that people should ask well before exceeding a timebox in more cases than not.

File: rules/ask-for-help/rule.md (hunk 7 above)
Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: I'm unsure if I like this. 

I think that a reword is fine. However, I do believe that someone should have an actionable piece of advice that would allow them to ask for help during the allotted effort. Otherwise, we're technically advising them to not ask for help unless the allotted effort is is exceeded. Which is not right.

File: rules/ask-for-help/rule.md [hunk 8]
```diff
@@ -4,14 +4,35 @@ title: Do you know when to ask for help?
...
 related:
   - do-you-use-timeboxing-to-avoid-wasted-time
+  - do-you-manage-up
+  - do-you-add-context-reasoning-to-your-emails
 created: 2022-08-30T04:06:01.553Z
 guid: 113b1e11-6df4-4901-9e9f-bcc06b6fb352
 ---
 When you're working, you will encounter issues that block you from progressing, force you to do a lot of research or otherwise spend time building towards being able to complete your work. In these moments, the question arises, "When should I ask for help?"
 
+Before asking for help, make sure you've done your due-diligence and exhausted your normal avenues for unblocking yourself. This concept is critical when trying to talk with Senior developers because their time is much more valuable. If you [call them without having done any research, it will waste a lot of their time and yours](https://www.ssw.com.au/rules/investigate-your-question-for-two-minutes-before-asking-someone-on-im) and won't be a good [ROI](https://www.ssw.com.au/rules/return-on-investment). Here are some examples for developers:
```
[the remark this answers is not in the document]
Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: Done - just one new line as adding a double carriage return in markdown is hard and not advisable.

File: rules/ask-for-help/rule.md (hunk 7 above)
Comment (pierssinclairssw) [reply rules/ask-for-help/rule.md]: Happy for it to be within the allocated effort, just want it to be easy to understand (aka simplify the words)

Comment (SamBWagner) [reply rules/ask-for-help/rule.md]: 100%. I'll re-word it and see if you like it :)

**Review by bradystroud — APPROVED** [review]

**Comment by bradystroud** [comment]
> Could we accept the rule 'as-is' for now so Adam can see the rule on the site during the meeting then re-open the comments after?
> 
> 
> 
> If no then please provide an alternative method for letting Adam see the rule. (Inside Netlify maybe?)

@SamBWagner 

Approved for now.

**Review by JackDevAU — APPROVED** [review]
